src/uu/ls/src/ls.rs

```diff
@@ -3195,17 +3185,58 @@ impl StyleManager {
     }
 }
 
-/// Colors the provided name based on the style determined for the given path.
+fn apply_style_based_on_metadata(
+    path: &PathData,
+    md_option: Option<&Metadata>,
+    ls_colors: &LsColors,
+    style_manager: &mut StyleManager,
+    name: &str,
+) -> String {
+    match ls_colors.style_for_path_with_metadata(&path.p_buf, md_option) {
+        Some(style) => style_manager.apply_style(style, name),
+        None => name.to_owned(),
+    }
+}
+
+/// Colors the provided name based on the style determined for the given path
+/// This function is quite long because it tries to leverage DirEntry to avoid
+/// unnecessary calls to stat()
+/// and manages the symlink errors
 fn color_name(
     name: String,
-    path: &Path,
-    md: Option<&Metadata>,
+    path: &PathData,
     ls_colors: &LsColors,
     style_manager: &mut StyleManager,
+    out: &mut BufWriter<Stdout>,
+    check_for_deref: bool,
```

Comment:
`check_for_deref` seems unnecessary to me because its value is linked to `target_symlink`: it's `true` if `target_symlink` is `Some(..)`, and `false`, if `target_symlink` is `None`. And as `color_name` is only called twice, it might make sense to split the function into two functions.

Reply:
wahou, well spotted, bravo :)